src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -12,16 +14,52 @@ namespace Microsoft.Extensions.Logging.Console
     [UnsupportedOSPlatform("browser")]
     internal class ConsoleLoggerProcessor : IDisposable
     {
-        private const int _maxQueuedMessages = 1024;
+        private const string WarningMessageOnDrop = " message(s) dropped because of queue size limit. Increase the queue size or decrease logging verbosity to avoid this. You may change `ConsoleLoggerBufferFullMode` to stop dropping messages.";
+        private readonly Queue<LogMessageEntry> _messageQueue;
+        private int _messagesDropped;
+        private bool _isAddingCompleted;
+        private int _maxQueuedMessages = ConsoleLoggerOptions.DefaultMaxQueueLengthValue;
+        public int MaxQueueLength
+        {
+            get => _maxQueuedMessages;
+            set
+            {
+                if (value <= 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} must be larger than zero.");
+                }
+
+                lock (_messageQueue)
+                {
+                    _maxQueuedMessages = value;
+                    Monitor.PulseAll(_messageQueue);
+                }
+            }
+        }
+        private ConsoleLoggerBufferFullMode _fullMode = ConsoleLoggerBufferFullMode.Wait;
+        public ConsoleLoggerBufferFullMode FullMode
+        {
+            get => _fullMode;
+            set
+            {
+                if (value != ConsoleLoggerBufferFullMode.Wait && value != ConsoleLoggerBufferFullMode.DropWrite)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, $"{nameof(value)} is not a supported buffer mode value.");
+                }
 
-        private readonly BlockingCollection<LogMessageEntry> _messageQueue = new BlockingCollection<LogMessageEntry>(_maxQueuedMessages);
+                _fullMode = value;
```

Comment:
Does this need a `lock` and `Monitor.PulseAll` like `MaxQueueLength` has?

Reply:
In theory it should.  Otherwise, you could have a bunch of threads all blocked waiting for room to open up in the queue, and if the full mode is then changed to drop, all of those threads should wake up, throw their data away, and exit.